lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -0,0 +1,424 @@
+
+    @staticmethod
+    def test_load_nsp(monkeypatch):
+        class MyNoSuchProcess:
+            def __init__(self, pid: int):
+                raise psutil.NoSuchProcess(pid)
+
+        mydir = AnotherPath("fake_dir", "12345")
+        ps = kill.PidSource("that.pid", "display that")
+        monkeypatch.setattr(
+            "pbench.cli.agent.commands.tools.kill.psutil.Process", MyNoSuchProcess
+        )
+        assert ps.load(mydir, "uuid1") is False
+
+    @staticmethod
+    def test_load_found(monkeypatch):
+        mydir = AnotherPath("fake_dir", "67890")
+        ps = kill.PidSource("that.pid", "display that")
+        monkeypatch.setattr(
+            "pbench.cli.agent.commands.tools.kill.psutil.Process", MyProcess
+        )
+        assert ps.load(mydir, "uuid1") is True
+
+    @staticmethod
+    def test_noop_killem(monkeypatch):
+        mock_kf = MagicMock()
+        monkeypatch.setattr(kill, "kill_family", mock_kf)
+        ps = kill.PidSource("this.pid", "display this")
+        ps.killem(ourecho)
+        assert not mock_kf.called
+
+    @staticmethod
+    def test_killem(monkeypatch):
+        mock_kf = MagicMock()
+        monkeypatch.setattr(kill, "kill_family", mock_kf)
+        monkeypatch.setattr(
+            "pbench.cli.agent.commands.tools.kill.psutil.Process", MyProcess
+        )
+        ps = kill.PidSource("this.pid", "display this")
+        ps.load(AnotherPath("fake_dir1", "123"), "uuid1")
+        ps.load(AnotherPath("fake_dir2", "456"), "uuid2")
+        ps.load(AnotherPath("fake_dir3", "789"), "uuid3")
+        ps.killem(ourecho)
+        assert len(mock_kf.mock_calls) == 3
+        assert mock_kf.mock_calls[0][1][0].pid == 123
+        assert mock_kf.mock_calls[1][1][0].pid == 456
+        assert mock_kf.mock_calls[2][1][0].pid == 789
+
+
+class TestKillTools:
+    @staticmethod
+    def test_gen_run_directories():
+        class FakePbenchRun:
+            """A very specific mock for a pbench_run Path object where we
+            leverage the AnotherPath class to yield a series of YAP objects
+            which respond as directories, passing along the value to be read
+            from a file they contain.
+            """
+
+            def __init__(self, dir_pairs: List[Tuple[str, str]]):
+                self.dir_names = dir_pairs
+
+            def iterdir(self):
+                for name, val in self.dir_names:
+                    yield AnotherPath(name, val)
+
+        pr = FakePbenchRun(
+            [
+                ("abc.file", None),  # top level file
+                ("filenotf", None),  # directory with no .uuid file
+                ("run1", "abcdef"),  # directory with a file
+                ("run2", "ghijkl"),  # directory with a file
+            ]
+        )
+        pairs = list(kill.gen_run_directories(pr))
+        assert pairs[0][0].name == "run1/tm" and pairs[0][1] == "abcdef"
+        assert pairs[1][0].name == "run2/tm" and pairs[1][1] == "ghijkl"
```

Comment:
If `pairs[0][0]` is a `pathlib.Path`, then it's really unlikely that `.name` would have a slash in it.  (I mean, it _could_ if the slash were escaped when the file was created, but it seems more likely to me that, something weird is going on with the code.)

Reply:
The current version of `AnotherPath` as reviewed here does.  I have updated the class to fix that.